src/ir_York.cpp

```diff
@@ -0,0 +1,358 @@
+#endif  // DECODE_YORK
+
+//
+//
+/// Class constructor
+/// @param[in] pin GPIO to be used when sending.
+/// @param[in] inverted Is the output signal to be inverted?
+/// @param[in] use_modulation Is frequency modulation to be used?
+IRYork::IRYork(const uint16_t pin, const bool inverted,
+                         const bool use_modulation)
+      : _irsend(pin, inverted, use_modulation) { }
+
+/// Set up hardware to be able to send a message.
+void IRYork::begin(void) { _irsend.begin(); }
+
+/// Get the raw state of the object, suitable to be sent with the appropriate
+/// IRsend object method.
+/// @return A copy of the internal state.
+uint8_t *IRYork::getRaw(void) {
+  return _.raw;
+}
+
+/// Set the internal state from a valid code for this protocol.
+/// @param[in] new_code A valid code for this protocol.
+/// @param[in] length Length of the code in bytes.
+void IRYork::setRaw(const uint8_t new_code[], const uint16_t length) {
+  std::memcpy(_.raw, new_code, length);
+}
+
+#if SEND_YORK
+/// Send the current internal state as an IR message.
+/// @param[in] repeat Nr. of times the message will be repeated.
+void IRYork::send(const uint16_t repeat) {
+  _irsend.sendYork(getRaw(), kYorkBits, repeat);
+}
+#endif  // SEND_YORK
+
+/// Get the current operation mode setting.
+/// @return The current operation mode.
+uint8_t IRYork::getMode(void) const {
+  return _.Mode;
+}
+
+/// Set the desired operation mode.
+/// @param[in] mode The desired operation mode.
+void IRYork::setMode(const uint8_t mode) {
+  switch (mode) {
+    case kYorkFan:
+    case kYorkCool:
+    case kYorkHeat:
+    case kYorkDry:
+    case kYorkAuto:
+      _.Mode = mode;
+      break;
+    default:
+      _.Mode = kYorkAuto;
+  }
+  setFan(getFan());  // Ensure the fan is at the correct speed for the new mode.
+}
+
+/// Convert a stdAc::opmode_t enum into its native mode.
+/// @param[in] mode The enum to be converted.
+/// @return The native equivalent of the enum.
+uint8_t IRYork::convertMode(const stdAc::opmode_t mode) {
+  switch (mode) {
+    case stdAc::opmode_t::kCool: return kYorkCool;
+    case stdAc::opmode_t::kHeat: return kYorkHeat;
+    case stdAc::opmode_t::kDry:  return kYorkDry;
+    case stdAc::opmode_t::kFan:  return kYorkFan;
+    default:                     return kYorkAuto;
+  }
+}
+
+/// Convert a native mode into its stdAc equivalent.
+/// @param[in] mode The native setting to be converted.
+/// @return The stdAc equivalent of the native setting.
+stdAc::opmode_t IRYork::toCommonMode(const uint8_t mode) {
+  switch (mode) {
+    case kYorkCool: return stdAc::opmode_t::kCool;
+    case kYorkHeat: return stdAc::opmode_t::kHeat;
+    case kYorkDry:  return stdAc::opmode_t::kDry;
+    case kYorkFan:  return stdAc::opmode_t::kFan;
+    default:           return stdAc::opmode_t::kAuto;
+  }
+}
+
+/// Set the speed of the fan.
+/// @param[in] speed The desired setting.
+/// @note The speed is locked to Low when in Dry mode and to auto when in auto mode.
```

Comment:
Your comment here doesn't match the code.
You're locking the result to Low when it is in `kYorkFan` mode.

Reply:
Great catch! Thanks a lot.